### PR TITLE
Collapse redundant types in the server part 2

### PR DESCRIPTION
**Pull request by jaredpar**
Follow up to #308411

This is probably best reviewed commit by commit. 

Note: I've intentionally avoided re-naming files because I want to keep the history diff as easy as possible. Once this is merged I will have one final PR which does all of the proper file renaming to match the new class names.

### CODE REVIEW

**Comment by jaredpar** [comment]
@dotnet/roslyn-compiler PTAL

File: src/Compilers/Server/VBCSCompilerTests/CompilerServerApiTest.cs [hunk 1]
```diff
@@ -302,6 +302,32 @@ public void ClientExceptionShouldBeginShutdown()
...
+            var host = new TestableCompilerServerHost(runCompilation: (request, cancellationToken) =>
+            {
+                buildStartedMre.Set();
+                clientClosedMre.WaitOne();
+                return new CompletedBuildResponse(0, utf8output: false, "");
+            });
+
+            using var serverData = await ServerUtil.CreateServer(compilerServerHost: host).ConfigureAwait(false);
+
+            // Create a short lived client that send a request but does not wait for the
```
Comment (cston) [inline src/Compilers/Server/VBCSCompilerTests/CompilerServerApiTest.cs]: >for the [](start = 81, length = 7)

"... response"?

File: src/Compilers/Shared/NamedPipeUtil.cs [hunk 2]
```diff
@@ -77,12 +77,12 @@ internal static bool CheckClientElevationMatches(NamedPipeServerStream pipeStrea
...
         /// Create a server for the current user only
         /// </summary>
-        internal static NamedPipeServerStream CreateServer(string pipeName)
+        internal static NamedPipeServerStream CreateServer(string pipeName, PipeDirection? pipeDirection = null)
         {
             var pipeOptions = PipeOptions.Asynchronous | PipeOptions.WriteThrough;
             return CreateServer(
                 pipeName,
-                PipeDirection.InOut,
+                pipeDirection ?? PipeDirection.InOut,
```
Comment (RikkiGibson) [inline src/Compilers/Shared/NamedPipeUtil.cs]: seems weird to use a nullable value type instead of using PipeDirection.InOut as the default value..?

Comment (jaredpar) [reply src/Compilers/Shared/NamedPipeUtil.cs]: That's the general pattern I've used for defaults in the server. Didn't want to break the trend here.

**Pull request merged**